lib/game.rb

```diff
@@ -31,123 +35,102 @@ def self.count_of_games_by_season
 
 #deliverable
   def self.average_goals_per_game
-    sum = @@all.sum { |game| game.away_goals + game.home_goals}.to_f
-    (sum / @@all.length.to_f).round(2)
-  end
-
-  def self.games_per(csv_header)
-    #returns number of games in a given (:season) or (:away_team_id)...
-    group_by_header = @@all.group_by { |game| game.send(csv_header) }
-    group_by_header.values.map{ |games| games.length}
-  end
-
+    sum = all.sum { |game| game.away_goals + game.home_goals}.to_f
+    (sum / all.length.to_f).round(2)
+  end
+
+  #MODULE!
+    def self.hash_of_hashes(collection, key1, key2, key3, value2, value3, arg2 = nil )
+      # {key1 => { key2 => value2, key3 => value3}} across collection. arg2 is ignored if not needed
+      # arg2 is optional if the passed method requires arguments
+      hash_of_hashes = Hash.new { |hash, key| hash[key] = {key2 => 0, key3 => 0}}
+      collection.each do |game|
+        hash_of_hashes[game.send(key1)][key2] += game.send(value2) if arg2.nil?
+        hash_of_hashes[game.send(key1)][key2] += game.send(value2, arg2) if !arg2.nil?
+        hash_of_hashes[game.send(key1)][key3] += value3
+      end
+      hash_of_hashes
+    end
 
-  def self.goals_per(csv_header, hoa_goals)
-    #returns number of goals (:home_goals) or (:away_goals) in a given csv_header
-    group_by_header = @@all.group_by { |game| game.send(csv_header) }
-    group_by_header.values.map do |games|
-      games.sum { |game| (game.send(hoa_goals))}
+  #MODULE!
+    def self.divide_hash_values(key1, key2, og_hash)
+      # accumulator hash {season => win%}
+      hash_divided = Hash.new { |hash, key| hash[key] = 0 }
+      # divide 2 hashe values and send to new hash value
+      og_hash.map do |key, value|
+        hash_divided[key] = (value[key1] / value[key2].to_f).round(2)
+      end
+      hash_divided
     end
```

Comment:
this helper method is GREAT for average or percentages. it divides the values created in the previous method.
if you start with 
```
{20132014=> { goals => 10, games_played => 5}
20142015=> { goals => 8, games_played => 3}}
```
after you divide `goals / games_played` the result is 
```
{20132014=> 2,
20142015=> 2.67}
```

Reply:
wow this seems very handy indeed